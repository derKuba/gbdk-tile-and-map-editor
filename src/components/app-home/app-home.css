.content {
  padding: 25px;
}

.content .center{
  display: flex;
  justify-content: center;
  margin-bottom: 25px;
}

.paint-field {
  margin-bottom: 25px;
  display: flex;
  justify-content: center;

}

.tool-box{
  margin-top: -8px;
}

.work-panel {
  display: flex;
  justify-content: center;
  margin-left: -295px;
}
