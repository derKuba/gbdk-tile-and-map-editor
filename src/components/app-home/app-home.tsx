import { Component, h } from '@stencil/core';

@Component({
    tag: 'app-home',
    styleUrl: 'app-home.css',
    shadow: true,
})
export class AppHome {
    render() {
        return (
            <div class="content">
                <mvjp-settings-ui />

                <div class="work-panel">
                    <mvjp-color-chooser></mvjp-color-chooser>
                    <mvjp-paint-modus-switcher />
                </div>

                <div class="paint-field">
                    <mvjp-canvas-matrix></mvjp-canvas-matrix>
                    <div class="tool-box">
                        <mvjp-preview size="s" />
                        <mvjp-preview-wall />
                        <mvjp-position-setting />
                        <mvjp-cp-panel />
                    </div>
                </div>

                <div class="center">
                    <mvjp-sprite-number-switcher></mvjp-sprite-number-switcher>
                </div>

                <div class="center">
                    <mvjp-export-button></mvjp-export-button>
                    <mvjp-modal>
                        <mvjp-export-modal-content />
                    </mvjp-modal>
                </div>
            </div>
        );
    }
}
