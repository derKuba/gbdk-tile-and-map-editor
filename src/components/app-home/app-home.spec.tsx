import { newSpecPage } from '@stencil/core/dist/testing';
import { h } from '@stencil/core';

import { AppHome } from './app-home';

describe('app', () => {
    it('builds', () => {
        expect(new AppHome()).toBeTruthy();
    });

    it('matches snapshot', async () => {
        const page = await newSpecPage({
            components: [AppHome],
            template: () => <app-home />,
        });

        expect(page.root).toMatchSnapshot();
    });
});
