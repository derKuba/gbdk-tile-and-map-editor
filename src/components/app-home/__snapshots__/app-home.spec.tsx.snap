// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app matches snapshot 1`] = `
<app-home>
  <mock:shadow-root>
    <div class="content">
      <mvjp-settings-ui></mvjp-settings-ui>
      <div class="work-panel">
        <mvjp-color-chooser></mvjp-color-chooser>
        <mvjp-paint-modus-switcher></mvjp-paint-modus-switcher>
      </div>
      <div class="paint-field">
        <mvjp-canvas-matrix></mvjp-canvas-matrix>
        <div class="tool-box">
          <mvjp-preview size="s"></mvjp-preview>
          <mvjp-preview-wall></mvjp-preview-wall>
          <mvjp-position-setting></mvjp-position-setting>
          <mvjp-cp-panel></mvjp-cp-panel>
        </div>
      </div>
      <div class="center">
        <mvjp-sprite-number-switcher></mvjp-sprite-number-switcher>
      </div>
      <div class="center">
        <mvjp-export-button></mvjp-export-button>
        <mvjp-modal>
          <mvjp-export-modal-content></mvjp-export-modal-content>
        </mvjp-modal>
      </div>
    </div>
  </mock:shadow-root>
</app-home>
`;
